src/runtime/nonce.zig

```diff
@@ -0,0 +1,67 @@
+const std = @import("std");
+const sig = @import("../sig.zig");
+
+const Hash = sig.core.Hash;
+const Pubkey = sig.core.Pubkey;
+const FeeCalculator = sig.runtime.sysvar.Fees.FeeCalculator;
+
+/// [agave] https://github.com/anza-xyz/agave/blob/faea52f338df8521864ab7ce97b120b2abb5ce13/sdk/program/src/nonce/state/current.rs#L10-L11
+const DURABLE_NONCE_HASH_PREFIX = "DURABLE_NONCE";
+
+/// Current variants have durable nonce and blockhash domains separated.\
+///
+/// Must support `bincode` and `serializedSize` methods for writing to the account data.\
+///
+/// [agave] https://github.com/anza-xyz/agave/blob/faea52f338df8521864ab7ce97b120b2abb5ce13/sdk/program/src/nonce/state/mod.rs#L12
+pub const Versions = union(enum) {
```

Comment:
Agave's implementation contains a couple more methods not implemented here? Is the idea to implement those later? if so maybe a TODO would be nice